### PR TITLE
feat: update stats streams to use new stats endpoints

### PR DESCRIPTION
**Pull request by jitoquinto**
- Creates some new classes with modified logic to get stats from the new `delivery` endpoint. 
- Updates incremental stream to use `IncrementalMixin` since other strategy is deprecated.

Also, this is gonna require some rework of the dbt we are using (schema changes to most streams)

### CODE REVIEW

File: source_stackadapt/streams.py [hunk 1]
```diff
@@ -260,101 +185,126 @@ def path(
...
-
-        return current_stream_state
-
-    def stream_slices(self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None) -> Iterable[Optional[Mapping[str, Any]]]:
+        return "total"
+    
+    @property
+    def date_range_type(self):
+        """
+        Override to specify the
```
Comment (kanelouise) [inline source_stackadapt/streams.py:238]: Here would we override to specify the end date?

**Review by kanelouise — COMMENTED** [review]
Hi, @jitoquinto! Please take my feedback with a grain of salt as you are far more expert in Airbyte than I am. 

First, I really appreciate the comments throughout the code. I think this will make it very easy for another person to understand and pickup this work.

There was one comment string that didn't seem like it was quite complete. I commented on that line directly. Might that be worth completing? My only other question is: is it whether it a red flag that tests are not passing?

File: source_stackadapt/streams.py (hunk 1 above)
Comment (jitoquinto) [reply source_stackadapt/streams.py:238]: Oh nice! thanks for catching! Ill update the doc on it

**Comment by jitoquinto** [comment]
> Hi, @jitoquinto! Please take my feedback with a grain of salt as you are far more expert in Airbyte than I am.
> 
> First, I really appreciate the comments throughout the code. I think this will make it very easy for another person to understand and pickup this work.
> 
> There was one comment string that didn't seem like it was quite complete. I commented on that line directly. Might that be worth completing? My only other question is: is it whether it a red flag that tests are not passing?

By tests do you mean the failed checks? It seems to be a perm issue when publishing the image to our GH container registry. This is the first time im trying to publish this connector there so still working out the issue! But yeah I'd like to get that resolved before merging in. It isnt a complete blocker since we've just been hosting it on my [personal dockerhub](https://hub.docker.com/r/jitoquinto/source-stackadapt/tags) account and can still do that, but I'd like to have it stored with our other [packages](https://github.com/community-tech-alliance/airbyte-source-stackadapt)

**Review by kanelouise — APPROVED** [review]
Ahh yes, @jitoquinto I was talking about the failed checked. But I see that it passes now! Nice work!

**Pull request merged**